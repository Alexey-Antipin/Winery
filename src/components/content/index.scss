.content {
  .content__inner {
    position: absolute;
    top: 295px;
    left: 179px;
    height: 472px;
    width: 1200px;
    .grape {
      position: absolute;
      right: -130px;
      bottom: -80px;
      height: 878px;
      width: 660px;
      background: url("../../image/grape.png");
      background-repeat: no-repeat;
      z-index: -1;
    }
    .content__img-line {
      margin-top: 20px;
      margin-bottom: 25px;
    }
    .content__text:nth-child(1) {
      font-family: "Playfair Display";
      font-style: normal;
      font-weight: 700;
      font-size: 50px;
      line-height: 67px;
      letter-spacing: 0.06em;
      color: #3f4042;
    }
    .content__text:nth-child(2) {
      font-family: "Playfair Display";
      font-style: normal;
      font-weight: 700;
      font-size: 70px;
      line-height: 93px;
      letter-spacing: 0.06em;
      color: #3f4042;
    }
    .content__text:nth-child(4) {
      max-width: 525px;
      font-family: "Avenir";
      font-style: normal;
      font-weight: 400;
      font-size: 18px;
      line-height: 147.6%;
      letter-spacing: 0.06em;
      color: #3f4042;
    }
    .content__block {
      height: 64px;
      width: 197px;
      position: relative;
      border: 2px solid #c08d5d;
      border-right: 59px solid #c08d5d;
      margin-top: 72px;
      .content__button {
        text-align: center;
        font-family: "Avenir";
        font-style: normal;
        font-weight: 400;
        font-size: 20px;
        line-height: 147.6%;
        letter-spacing: 0.06em;
        padding: 15px;
        background: #fffaf3;
      }
      .content__img {
        position: absolute;
        top: 25px;
        right: -40px;
      }
    }
  }
  .content-logo {
    margin: 0px;
  }
}

@media (max-width: 800px) {
  .content {
    .content__inner {
      position: static;
      width: 600px;
      padding: 20px;
      margin: 0 auto;
      .content__text:nth-child(4) {
        max-width: 350px;
      }
      .grape {
        position: relative;
        top: -580px;
        right: -240px;
        height: 878px;
        width: 660px;
      }
    }
  }
}
