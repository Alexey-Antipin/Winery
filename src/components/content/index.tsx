import { vector, millet, line } from "../../image";
import "./index.scss";

export const Content: React.FC = () => {
  return (
    <div className="content">
      <div className="content__inner">
        <div className="content__text">
          Food
          <img className="logo content-logo" src={millet} alt="millet"></img>
        </div>
        <div className="content__text">FARMING</div>
        <img className="content__img-line" src={line} alt="vector" />

        <div className="content__text">
          Food Farming is proud of our past and excited about our future. We
          strive to continue being one of India and New Zealand's great food
          companies, offering quality fruits and vegetable.
        </div>

        <div className="content__block">
          <div className="content__button">Read More</div>
          <img className="content__img" src={vector} alt="vector" />
        </div>

        <div className="grape"/>
      </div>
    </div>
  );
};
