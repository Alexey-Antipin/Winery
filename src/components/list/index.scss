.list {
  display: flex;
  justify-content: space-around;
  width: 305px;
  .text {
    font-style: normal;
    font-weight: 500;
    font-size: 20px;
    line-height: 27px;
    list-style-type: none;
    color: #000000;
  }
}
