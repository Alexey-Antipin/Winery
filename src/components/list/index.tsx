import "./index.scss";

type Props = {
  massive: Massive[];
  cls?: string;
};

type Massive = {
  text: string;
  href: string;
};

export const List: React.FC<Props> = ({ massive, cls }) => {
  return (
    <ul className="list">
      {massive.map((item, index) => (
        <li className={`text ${index === 0 && cls}`} key={index}>
          {item.text}
        </li>
      ))}
    </ul>
  );
};
