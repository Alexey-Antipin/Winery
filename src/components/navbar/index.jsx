import React from "react";
import millet from "../../image/millet.png";
import box from "../../image/box.png";
import "./index.scss";

export const Navbar = () => {
  return (
    <div className="navbar">
      <div className="navbar__inner">
        <img className="navbar__box" src={box} alt="box" />
        <div className="navbar__container-list">
          <ul className="navbar__list">
            <li className="navbar__text navbar__text--active">Home</li>
            <li className="navbar__text">About</li>
            <li className="navbar__text">Products</li>
          </ul>
          <img className="logo logo-mobile" src={millet} alt="millet"></img>
          <ul className="navbar__list">
            <li className="navbar__text">Recipes</li>
            <li className="navbar__text">FAQ</li>
            <li className="navbar__text">Help</li>
          </ul>
        </div>
      </div>
    </div>
  );
};
