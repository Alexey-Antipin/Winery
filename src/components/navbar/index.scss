.navbar {
  .navbar__inner {
    display: flex;
    justify-content: center;
    align-items: center;
    height: 146px;
    max-width: 1460px;
    overflow-x: hidden;

    .navbar__box {
      padding-left: 20px;
      margin-right: auto;
    }
    .navbar__container-list {
      display: flex;
      align-items: center;
      width: 850px;
      margin-right: auto;
      .text--active {
        color: #c08d5d;
      }
    }
  }
}

.logo {
  height: 59px;
  width: 38.64px;
  margin: 0 auto;
}

@media (max-width: 700px) {
  .navbar {
    .navbar__inner {
      .navbar__container-list {
        flex-direction: column;
      }
      .logo-mobile {
        position: absolute;
        top: 40px;
        right: 30px;
        order: 1;
      }
    }
  }
}
