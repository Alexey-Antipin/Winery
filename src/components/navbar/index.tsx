import { millet, box } from "../../image";
import "./index.scss";

export const Navbar: React.FC = () => {
  return (
    <div className="navbar">
      <div className="navbar__inner">
        <img className="navbar__box" src={box} alt="box" />
        <div className="navbar__container-list">
          <ul className="navbar__list">
            <li className="navbar__text navbar__text--active">Home</li>
            <li className="navbar__text">About</li>
            <li className="navbar__text">Products</li>
          </ul>
          <img className="logo logo-mobile" src={millet} alt="millet"></img>
          <ul className="navbar__list">
            <li className="navbar__text">Recipes</li>
            <li className="navbar__text">FAQ</li>
            <li className="navbar__text">Help</li>
          </ul>
        </div>
      </div>
    </div>
  );
};
