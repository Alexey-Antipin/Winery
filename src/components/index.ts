import { Content } from "./content";
import { Navbar } from "./navbar";
import { List } from "./list";

export { Navbar, Content, List };
