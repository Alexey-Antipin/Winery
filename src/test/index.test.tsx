import { render } from "@testing-library/react";
import { Navbar, Content } from "../components";
import { App } from "../app";

describe("Тестирование", () => {
  test("Все компоненты", () => {
    const view = render(<App />);
    expect(view).toMatchSnapshot();
  });

  test("Navbar", () => {
    const { container } = render(<Navbar />);
    // eslint-disable-next-line testing-library/no-container, testing-library/no-node-access
    const cls = container.getElementsByClassName("navbar");
    expect(cls.length).toBe(1);
  });
  test("Content", () => {
    const { container } = render(<Content />);
    // eslint-disable-next-line testing-library/no-container, testing-library/no-node-access
    const cls = container.getElementsByClassName("content");
    expect(cls.length).toBe(1);
  });
});
