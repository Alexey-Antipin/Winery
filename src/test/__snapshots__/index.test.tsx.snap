// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование Все компоненты 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="navbar"
        >
          <div
            class="navbar__inner"
          >
            <img
              alt="box"
              class="navbar__box"
              src="box.png"
            />
            <div
              class="navbar__container-list"
            >
              <ul
                class="navbar__list"
              >
                <li
                  class="navbar__text navbar__text--active"
                >
                  Home
                </li>
                <li
                  class="navbar__text"
                >
                  About
                </li>
                <li
                  class="navbar__text"
                >
                  Products
                </li>
              </ul>
              <img
                alt="millet"
                class="logo logo-mobile"
                src="millet.png"
              />
              <ul
                class="navbar__list"
              >
                <li
                  class="navbar__text"
                >
                  Recipes
                </li>
                <li
                  class="navbar__text"
                >
                  FAQ
                </li>
                <li
                  class="navbar__text"
                >
                  Help
                </li>
              </ul>
            </div>
          </div>
        </div>
        <div
          class="content"
        >
          <div
            class="content__inner"
          >
            <div
              class="content__text"
            >
              Food
              <img
                alt="millet"
                class="logo content-logo"
                src="millet.png"
              />
            </div>
            <div
              class="content__text"
            >
              FARMING
            </div>
            <img
              alt="vector"
              class="content__img-line"
              src="vectorLine.png"
            />
            <div
              class="content__text"
            >
              Food Farming is proud of our past and excited about our future. We strive to continue being one of India and New Zealand's great food companies, offering quality fruits and vegetable.
            </div>
            <div
              class="content__block"
            >
              <div
                class="content__button"
              >
                Read More
              </div>
              <img
                alt="vector"
                class="content__img"
                src="vector.png"
              />
            </div>
            <div
              class="grape"
            />
          </div>
        </div>
      </div>
      <div
        class="mouse"
      >
        <div
          class="mouse__point"
        />
        <div
          class="mouse__point"
        />
        <img
          alt="mouse"
          class="mouse__img"
          src="mouse.png"
        />
      </div>
      <footer
        class="footer"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="navbar"
      >
        <div
          class="navbar__inner"
        >
          <img
            alt="box"
            class="navbar__box"
            src="box.png"
          />
          <div
            class="navbar__container-list"
          >
            <ul
              class="navbar__list"
            >
              <li
                class="navbar__text navbar__text--active"
              >
                Home
              </li>
              <li
                class="navbar__text"
              >
                About
              </li>
              <li
                class="navbar__text"
              >
                Products
              </li>
            </ul>
            <img
              alt="millet"
              class="logo logo-mobile"
              src="millet.png"
            />
            <ul
              class="navbar__list"
            >
              <li
                class="navbar__text"
              >
                Recipes
              </li>
              <li
                class="navbar__text"
              >
                FAQ
              </li>
              <li
                class="navbar__text"
              >
                Help
              </li>
            </ul>
          </div>
        </div>
      </div>
      <div
        class="content"
      >
        <div
          class="content__inner"
        >
          <div
            class="content__text"
          >
            Food
            <img
              alt="millet"
              class="logo content-logo"
              src="millet.png"
            />
          </div>
          <div
            class="content__text"
          >
            FARMING
          </div>
          <img
            alt="vector"
            class="content__img-line"
            src="vectorLine.png"
          />
          <div
            class="content__text"
          >
            Food Farming is proud of our past and excited about our future. We strive to continue being one of India and New Zealand's great food companies, offering quality fruits and vegetable.
          </div>
          <div
            class="content__block"
          >
            <div
              class="content__button"
            >
              Read More
            </div>
            <img
              alt="vector"
              class="content__img"
              src="vector.png"
            />
          </div>
          <div
            class="grape"
          />
        </div>
      </div>
    </div>
    <div
      class="mouse"
    >
      <div
        class="mouse__point"
      />
      <div
        class="mouse__point"
      />
      <img
        alt="mouse"
        class="mouse__img"
        src="mouse.png"
      />
    </div>
    <footer
      class="footer"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
